app/domain/authentication/authn_iam/authenticator.rb

```diff
@@ -54,13 +54,34 @@ def extract_relevant_data(response)
 
       # Call to AWS STS endpoint using the provided authentication header
       def attempt_signed_request(signed_headers)
-        aws_request = URI("https://#{signed_headers['host']}/?Action=GetCallerIdentity&Version=2011-06-15")
-        begin
-          @client.get_response(aws_request, signed_headers)
+        region = extract_sts_region(signed_headers)
+
+        # Attempt request using the discovered region and return immediately if successful
+        response = aws_call(region: region, headers: signed_headers)
+        return response if response.code.to_i == 200
+      
+        # If the discovered region is `us-east-1`, fallback to the global endpoint
+        if region == 'us-east-1'
+          @logger.debug(LogMessages::Authentication::AuthnIam::RetryWithGlobalEndpoint.new)
+          fallback_response = aws_call(region: 'global', headers: signed_headers)
+          return fallback_response if fallback_response.code.to_i == 200
+        end
 
-          # Handle any network failures with a generic verification error
+        return response
```

Comment:
Wonder if in the case where `region == 'us-east-1'` and `fallback_response.code.to_i != 200` we want to return the `fallback_response` instead of the primary `response`.

Reply:
I wasn't too sure about this. There isn't a way to definitively know whether the regional endpoint or the global endpoint was used to generate the signed headers even after attempting both requests. AWS throws a fairly generic `The request signature we calculated does not match the signature you provided` error message either way